src/handlers/oidc.rs

```diff
@@ -115,21 +118,48 @@ broker_handler!(Auth, |app, req| {
     }
 
     if app.providers.contains_key(&email_addr.domain.to_lowercase()) {
-
         // OIDC authentication. Redirect to the identity provider.
         let auth_url = oidc_bridge::request(app, email_addr, client_id, nonce, &parsed_redirect_uri)?;
         Ok(Response::with((status::SeeOther, modifiers::Header(Location(auth_url.to_string())))))
 
     } else {
+        let langtag_de: LanguageTag = "de".parse().unwrap();
+        let mut langtag_en: LanguageTag = Default::default();
+        langtag_en.language = Some("en".to_owned());
+        let langtags = [langtag_de, langtag_en];
+        let catalog_files = ["lang/german.mo", "lang/english.mo"];
+
+        let mut langtag_en_: LanguageTag = Default::default();
+        langtag_en_.language = Some("en".to_owned());
+        let default_accept_language_header = AcceptLanguage(vec![qitem(langtag_en_),]);
+
+        let accept_language_header = req.headers.get::<AcceptLanguage>().unwrap_or(&default_accept_language_header);
+
+        let mut lang_file = "";
+        'outer: for accept_language in accept_language_header.iter() {
+            for (index, langtag) in langtags.iter().enumerate() {
+                if langtag.matches(&accept_language.item) {
+                    lang_file = catalog_files[index];
+                    break 'outer;
+                }
+            }
+        }
+        
+        let f = File::open(lang_file).expect("could not open the catalog");
+        let catalog = Catalog::parse(f).expect("could not parse the catalog");
```

Comment:
Thanks a lot @stephank. I pushed https://github.com/portier/portier-broker/pull/124/commits/7a43105d628d3f11f94b0347d0fb0360e23401a4, which is a mix of our two approaches. I'm very happy about the improvements you could show me.

To your comments:

 * I changed that to to_owned, but later on was able to remove that part completely
 * `Clone` is way better here, I searched that
 * Agreed and changed
 * We probably should, but I don't know how to implement that. Have a static(?) Vec<LanguageTag, Catalog> in config that is filled on init? That's too much rust for me. I instead adapted your code to only create one catalogue, the needed one.
 * Yes, that was pretty unnecessary. I did not understand how to use `if let Some(accept)`.

Could we merge this now? Or does someone want to move the catalogue parsing to init? I think that would be nice, but not something that should block this PR if no one wants to do it immediately. We can open an issue for  that.